src/services/db/__tests__/GitFileCommitService.spec.ts

```diff
@@ -0,0 +1,555 @@
+import { GitHubCommitData } from "@root/types/commitData"
+import { GitDirectoryItem, GitFile } from "@root/types/gitfilesystem"
+import _GitFileCommitService from "@services/db/GitFileCommitService"
+import _GitFileSystemService from "@services/db/GitFileSystemService"
+
+import * as gbUtils from "../../../utils/growthbook-utils"
+
+const MockSimpleGit = {
+  clone: jest.fn(),
+  cwd: jest.fn(),
+}
+
+const gitFileSystemService = new _GitFileSystemService(
+  (MockSimpleGit as unknown) as SimpleGit
+)
+
+const gitFileCommitService = new _GitFileCommitService(gitFileSystemService)
+
+const BRANCH_REF = config.get("github.branchRef")
+const DEFAULT_BRANCH = "staging"
+const sessionData = mockUserWithSiteSessionData
+describe("GitFileCommitService", () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("pushToGithub", () => {
+    it("should push to staging and staging-lite", async () => {
+      // Arrange
+      const pushSpy = jest
+        .spyOn(gitFileSystemService, "push")
+        .mockImplementation((siteName, branchName) =>
+          ResultAsync.fromPromise(
+            Promise.resolve("push ok"),
+            () => new GitFileSystemError("push failed")
+          )
+        )
+
+      // Act
+      await gitFileCommitService.pushToGithub(sessionData, true)
+
+      // Assert
+      expect(pushSpy).toHaveBeenCalledWith(sessionData.siteName, STAGING_BRANCH)
+      expect(pushSpy).toHaveBeenCalledWith(
+        sessionData.siteName,
+        STAGING_LITE_BRANCH
+      )
+    })
+
+    it("should only push to staging if shouldUpdateStagingLite is false", async () => {
+      // Arrange
+      const pushSpy = jest
+        .spyOn(gitFileSystemService, "push")
+        .mockResolvedValue(okAsync("push ok"))
+
+      // Act
+      await gitFileCommitService.pushToGithub(sessionData, false)
+
+      // Assert
+      expect(pushSpy).toHaveBeenCalledWith(sessionData.siteName, STAGING_BRANCH)
+      expect(pushSpy).toHaveBeenCalledTimes(1)
+    })
+
+    describe("create", () => {
+      it("should create a file and push to GitHub", async () => {
+        // Arrange
+
+        const content = "file content"
+        const fileName = "file.txt"
+        const directoryName = "directory"
+        const isMedia = false
+
+        const createSpy = jest
+          .spyOn(gitFileSystemService, "create")
+          .mockResolvedValue(okAsync({ newSha: "new-sha" }))
+        const pushSpy = jest
+          .spyOn(gitFileCommitService, "pushToGithub")
+          .mockResolvedValue()
+
+        // Act
+        const result = await gitFileCommitService.create(sessionData, {
+          content,
+          fileName,
+          directoryName,
+          isMedia,
+        })
+
+        // Assert
+        expect(createSpy).toHaveBeenCalledWith(
+          sessionData.siteName,
+          sessionData.isomerUserId,
+          content,
+          directoryName,
+          fileName,
+          isMedia ? "base64" : "utf-8",
```

Comment:
nit: why do we need conditionals here for a hardcoded value of isMedia?

Reply:
[33bff68](https://github.com/isomerpages/isomercms-backend/pull/1047/commits/33bff68ae09ef4af6831fb861834da5693fd4d97)